### ISSUE TITLE
error: externally-managed-environment

### ISSUE THREAD

**Issue by scottonanski** [comfyanonymous/ComfyUI#3554]
From the installation guide, running;

`pip install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu121`

Returns the following ;

```
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.
    
    If you wish to install a non-Debian packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.
    
    See /usr/share/doc/python3.12/README.venv for more information.

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
```

What do?

Thanks..

**Comment by scottonanski** [comment]
Figured it out.

I'm running a fresh install of Ubuntu 24.04, and this was my solution;

Step 1: 
`cd ~/AI\ Applications/ComfyUI` (Or wherever you have it installed...);

Step 2:
`sudo apt install python3.12-venv` (install python 3 virtual environment);

Step 3:
`source venv/bin/activate` (activate the virtual environment);

Step 4:
`pip install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu121` (required packages)

Step 5:
'pip install -r requirements.txt'

In a nutshell;

```
cd ~/AI\ Applications/ComfyUI
python3 -m venv venv
source venv/bin/activate
pip install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu121
pip install -r requirements.txt
```

Easy-peasy, lemon-squeezy.
